File: data/github/current/packages/site/app/models/site/contentful/marketing/newsroom/schemas/home_page.rb
# typed: true
# frozen_string_literal: true

module Site
  module Contentful
    module Marketing
      module Newsroom
        module Schemas
          module HomePage
            sig { returns(T::Hash[T.untyped, T.untyped]) }
            def self.build
              {
                "$schema": "http://json-schema.org/draft-04/schema#",
                "title": "newsroomTemplateHomepage",
                "type": "object",
                "properties": {
                  "hero": { "type": "object" },
                  "heroBackgroundImage": { "type": "object" },
                  "heroStatistics": {
                    "type": "array",
                    "minItems": 3,
                    "maxItems": 3,
                  },
                  "pressReleaseSectionIntro": { "type": "object" },
                  "pressReleaseSectionCards": { "type": "object" },
                  "reportsSectionCards": { "type": "object" },
                  "reportsSectionStatistics": {
                    "type": "array",
                    "minItems": 3,
                    "maxItems": 3,
                  },
                  "inTheNewsSectionIntro": { "type": "object" },
                  "inTheNewsSectionCards": { "type": "object" },
                  "ctaBanner": { "type": "object" },
                },
                "required": %w[hero heroBackgroundImage heroStatistics pressReleaseSectionIntro pressReleaseSectionCards reportsSectionCards reportsSectionStatistics inTheNewsSectionIntro inTheNewsSectionCards ctaBanner],
              }
            end
          end
        end
      end
    end
  end
end

File: data/github/current/ui/packages/newsroom/lib/types/contentful/contentTypes/templates/homepage.ts
import {z} from 'zod'
import {buildEntrySchemaFor} from '@github-ui/swp-core/schemas/contentful/entry'
import {PrimerComponentHeroSchema} from '@github-ui/swp-core/schemas/contentful/contentTypes/primerComponentHero'
import {BackgroundImageSchema} from '@github-ui/swp-core/schemas/contentful/contentTypes/backgroundImage'
import {PrimerComponentSectionIntroSchema} from '@github-ui/swp-core/schemas/contentful/contentTypes/primerComponentSectionIntro'
import {PrimerCardsSchema} from '@github-ui/swp-core/schemas/contentful/contentTypes/primerCards'
import {PrimerComponentCtaBannerSchema} from '@github-ui/swp-core/schemas/contentful/contentTypes/primerComponentCtaBanner'
import {PrimerComponentStatisticSchema} from '@github-ui/swp-core/schemas/contentful/contentTypes/primerComponentStatistic'
import {AssetSchema} from '@github-ui/swp-core/schemas/contentful/contentTypes/asset'

export const TemplateHomeSchema = buildEntrySchemaFor('newsroomTemplateHomepage', {
  fields: z.object({
    hero: PrimerComponentHeroSchema,
    heroBackgroundImage: BackgroundImageSchema,
    heroStatistics: z.array(PrimerComponentStatisticSchema).optional(),
    pressReleaseSectionIntro: PrimerComponentSectionIntroSchema,
    pressReleaseSectionCards: PrimerCardsSchema,
    reportsSectionIntro: PrimerComponentSectionIntroSchema,
    reportsSectionCards: PrimerCardsSchema,
    reportsStatisticsHeading: z.string().optional(),
    reportsSectionStatistics: z.array(PrimerComponentStatisticSchema).optional(),
    inTheNewsSectionIntro: PrimerComponentSectionIntroSchema,
    inTheNewsSectionCards: PrimerCardsSchema,
    customerStoriesSectionIntro: PrimerComponentSectionIntroSchema.optional(),
    featuredCustomerStories: PrimerCardsSchema.optional(),
    ctaSectionImage: AssetSchema.optional(),
    ctaSectionHeading: z.string().optional(),
    ctaSectionCards: PrimerCardsSchema.optional(),
    ctaBanner: PrimerComponentCtaBannerSchema,
    ctaBannerBackgroundImage: AssetSchema.optional(),
  }),
})

export type HomeTemplate = z.infer<typeof TemplateHomeSchema>
